pkg/api/v1/types.go

```diff
@@ -437,8 +440,21 @@ const (
 
 	// MountOptionAnnotation defines mount option annotation used in PVs
 	MountOptionAnnotation = "volume.beta.kubernetes.io/mount-options"
+
+	// StorageTopologyConstraint defines topology constraints of a PV
+	// Value is type []TopologyConstraint
+	AlphaStorageTopologyConstraintsAnnotation = "volume.alpha.kubernetes.io/topology-constraints"
 )
 
+// TopologyConstraint contains label information that the resource is constrained to
+// A label key can have multiple values
+type TopologyConstraint struct {
+	// Key of the label
```

Comment:
Even though the idea is to allow user to use any label, the type is named "Topology" and we do have 'first-class' [keys](https://kubernetes.io/docs/reference/labels-annotations-taints/).  I guess for the most of time, user will just use 'kubernetes.io/hostname'.